### PR TITLE
Use `custom_function` attribute when extracting custom metric function

### PR DESCRIPTION
**Pull request by harupy**
## Related Issues/PRs

 #xxx

## What changes are proposed in this pull request?

Use `custom_function` attribute when extracting custom metric function.

## How is this patch tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [x] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [x] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/recipes/utils/metrics.py [hunk 1]
```diff
@@ -150,7 +150,7 @@ def _load_custom_metrics(recipe_root: str, metrics: List[RecipeMetric]) -> List[
         custom_metrics_mod = importlib.import_module("steps.custom_metrics")
         return [
             make_metric(
-                eval_fn=getattr(custom_metrics_mod, custom_metric.name),
+                eval_fn=getattr(custom_metrics_mod, custom_metric.custom_function),
```
Comment (harupy) [inline mlflow/recipes/utils/metrics.py:153]: Found this issue when I was testing the classification template. We must use `custom_function`, not `name`.

```yaml
custom_metrics:
  # FIXME::OPTIONAL: Defines custom performance metrics to compute during model development.
    - name: mam
      function: my_awesome_metric_func
      greater_is_better: False
```

```python
def my_awesome_metric_func(). 
   return 1_000_000
```

Comment (WeichenXu123) [reply mlflow/recipes/utils/metrics.py:153]: Should it be `custom_metric.function` ?

Comment (harupy) [reply mlflow/recipes/utils/metrics.py:153]: `custom_metric` is a `RecipeMetric` object.

Comment (WeichenXu123) [reply mlflow/recipes/utils/metrics.py:153]: Got it.

Comment (harupy) [reply mlflow/recipes/utils/metrics.py:153]: If `custom_metric` was the correct one, tests would fail.

**Pull request merged**